### PR TITLE
Import asset in ImportScanMutation

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

**Review by amine3 — APPROVED** [review]

**Review by amine3 — DISMISSED** [review]
Unit tests fail.  LGTM

File: src/ostorlab/serve_app/common.py [hunk 1]
```diff
@@ -274,3 +276,31 @@ def is_aab(file_content: bytes) -> bool:
...
         return False
+
+
+def get_package_name(path: str) -> str:
+    """Get the package name of an android file."""
+    apk_info = pyaxmlparser.APK(path)
+    return apk_info.get_package()
+
+
+def get_bundle_id(ipa_path: str) -> str:
```
Comment (ohachimOs) [inline src/ostorlab/serve_app/common.py:287]: no docs for this?

Comment (elyousfi5) [reply src/ostorlab/serve_app/common.py]: added

File: src/ostorlab/serve_app/common.py [hunk 2]
```diff
@@ -274,3 +276,48 @@ def is_aab(file_content: bytes) -> bool:
...
+
+
+def get_bundle_id(path: str) -> str:
+    """Get the bundle identifier of an iOS file.
+
+    Args:
+        path: Path to the iOS file.
+
+    Returns:
+        str: Bundle identifier of the iOS file.
```
Comment (ErebusZ) [inline src/ostorlab/serve_app/common.py:302]:
```suggestion
        path: Path to the iOS ipa file.

    Returns:
        str: Bundle identifier of the iOS ipa file.
```

File: src/ostorlab/serve_app/common.py [hunk 3]
```diff
@@ -274,3 +276,48 @@ def is_aab(file_content: bytes) -> bool:
...
+                and dirs[2].lower().strip() == "info.plist"
+            ):
+                content = z.open(filename).read()
+                infoplist = plistlib.loads(content)
+
+        if infoplist is None:
+            raise ValueError("Info.plist not found in the IPA")
+
+        # Extract and read the Info.plist file
+        return infoplist.get("CFBundleIdentifier", "")
```
Comment (ErebusZ) [inline src/ostorlab/serve_app/common.py:323]: This should return None as we shoudl not create an IOSAsset without bundle id.

Comment (elyousfi5) [reply src/ostorlab/serve_app/common.py:323]: Thats true for android and ios store. But for android file and ios file its not the case !

File: src/ostorlab/serve_app/common.py [hunk 4]
```diff
@@ -274,3 +276,52 @@ def is_aab(file_content: bytes) -> bool:
...
+    """Get the package name of an android file.
+
+    Args:
+        path: Path to the android file.
+
+    Returns:
+        str: Package name of the android file.
+    """
+
+    apk_info = pyaxmlparser.APK(path)
```
[the remark this answers is not in the document]
Comment (elyousfi5) [reply src/ostorlab/serve_app/common.py:291]: nope, we do create the file in this path before calling get_package_name or get_bundle_id: 

image

**Review by 3asm — APPROVED** [review]

**Pull request merged**